ton_client/src/debot/sdk_interface.rs

```diff
@@ -294,6 +314,33 @@ pub struct SdkInterface {
     ton: TonClient,
 }
 
+#[derive(Default, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub(super) struct EncryptionBoxInfoResult {
+    pub hdpath: String,
+    pub algorithm: String,
+    pub options: String,
+    pub public_info: String,
+}
+
+use std::convert::From;
+impl From<EncryptionBoxInfo> for EncryptionBoxInfoResult {
+    fn from(info: EncryptionBoxInfo) -> Self {
+        Self {
+            algorithm: info.algorithm.map(|v| hex::encode(v)).unwrap_or_default(),
```

Comment:
is it correct: encoding optionals instead of strings, after optional unwrapping?

Reply:
First, we map value inside `Option` if it has `Some(...)` value. Then we extract value from Option if it's exists or return default String (empty).